File: java/src/main/java/Exch.java
import com.google.gson.Gson;
import com.google.gson.JsonArray;
import com.google.gson.JsonObject;
import com.nicehash.connect.Api;
import org.apache.commons.logging.Log;
import org.apache.commons.logging.LogFactory;

import java.util.UUID;

/**
 * exchange example
 */
public class Exch
{
	private Log log = LogFactory.getLog(Exch.class);

	private static final String URL_ROOT   = "https://api-test.nicehash.com/"; //use https://api2.nicehash.com for production
	private static final String ORG_ID     = "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee"; //get it here: https://test.nicehash.com/my/settings/keys or https://new.nicehash.com/my/settings/keys
	private static final String API_KEY    = "ffffffff-gggg-hhhh-iiii-jjjjjjjjjjjj";
	private static final String API_SECRET = "kkkkkkkk-llll-mmmm-nnnn-oooooooooooooooooooo-pppp-qqqq-rrrr-ssssssssssss";

	private static final String CURRENCY_SELL = "TBTC"; //user BTC for production
	private static final String CURRENCY_BUY  = "TLTC"; //use LTC for production

    public static void main( String[] args ) {
		new Exch();
    }

	private Exch() {
	    Api api = new Api(URL_ROOT, ORG_ID, API_KEY, API_SECRET);

	    //get server time
	    String timeResponse = api.get("api/v2/time");
		JsonObject timeObject = new Gson().fromJson(timeResponse, JsonObject.class);
		String time = timeObject.get("serverTime").getAsString();
	    log.info("server time: " + time);

		//get exchange settings
		String exchResponse = api.get("exchange/api/v2/info/status");
		JsonObject exchObject = new Gson().fromJson(exchResponse, JsonObject.class);
		//log.info("exchanges: " + exchObject.toString());

		final JsonObject[] exchanges = new JsonObject[1];
		exchObject.get("symbols").getAsJsonArray().forEach(exch-> {
			JsonObject e = exch.getAsJsonObject();
			if (e.get("baseAsset").getAsString().equals(CURRENCY_BUY)) {
				exchanges[0] = e;
			}
		});

		JsonObject mySettings = exchanges[0];
		log.info("exchange settings: " + mySettings);

		//get balance
		String activityResponse = api.get("main/api/v2/accounting/accounts", true, time);
		JsonArray accountsArray = new Gson().fromJson(activityResponse, JsonArray.class);
		log.info("accounts: " + accountsArray.toString());

		final double[] balance = new double[1];
		accountsArray.forEach(acc-> {
			JsonObject a = acc.getAsJsonObject();
			if (a.get("currency").getAsString().equals(CURRENCY_SELL)) {
				balance[0] = a.get("balance").getAsDouble();
			}
		});

		//get order book
		String orderBookResponse = api.get("exchange/api/v2/orderbook?market="+CURRENCY_BUY+CURRENCY_SELL+"&limit=100", true, time);
		JsonObject orderBookObject = new Gson().fromJson(orderBookResponse, JsonObject.class);
		log.info("order book: " + orderBookObject.toString());

		//cheapest offer for CURRENCY_BUY
		JsonArray sellArray = orderBookObject.get("sell").getAsJsonArray();
		JsonArray cheapestArray = sellArray.get(0).getAsJsonArray();
		log.info("cheapest offer price: " + cheapestArray.get(0) + " supply: " + cheapestArray.get(1));

		double qty = mySettings.get("secMinAmount").getAsDouble()*2;

		//buy with limit order
		String url = "exchange/api/v2/order?market="+CURRENCY_BUY+CURRENCY_SELL+"&side=buy&type=limit&quantity="+qty+"&price="+ cheapestArray.get(0);
		log.info("order url: " +url);
		String orderCreateResponse = api.post(url, null, time, true);
		JsonObject orderCreateObject = new Gson().fromJson(orderCreateResponse, JsonObject.class);
		log.info("order create: " + orderCreateObject.toString());


    }
}

File: nuxhash/nhrest/c#/connect/connect/Exch.cs
using NLog;
using System;
using System.Collections.Generic;
using System.Data;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace connect
{
    class Exch
    {
        private static NLog.Logger Logger = NLog.LogManager.GetCurrentClassLogger();

        private static string URL_ROOT   = "https://api-test.nicehash.com"; //use https://api2.nicehash.com for production
        private static string ORG_ID     = "aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee";
        private static string API_KEY    = "ffffffff-gggg-hhhh-iiii-jjjjjjjjjjjj";
        private static string API_SECRET = "kkkkkkkk-llll-mmmm-nnnn-oooooooooooooooooooo-pppp-qqqq-rrrr-ssssssssssss";

        private static string CURRENCY_SELL = "TBTC"; //user BTC for production
        private static string CURRENCY_BUY  = "TLTC"; //use LTC for production

        public Exch()
        {
            var config = new NLog.Config.LoggingConfiguration();
            var logconsole = new NLog.Targets.ConsoleTarget("logconsole");
            config.AddRule(LogLevel.Info, LogLevel.Fatal, logconsole);
            NLog.LogManager.Configuration = config;

            Api api = new Api(URL_ROOT, ORG_ID, API_KEY, API_SECRET);

            //get server time
            string timeResponse = api.get("/api/v2/time");
            ServerTime serverTimeObject = Newtonsoft.Json.JsonConvert.DeserializeObject<ServerTime>(timeResponse);
            string time = serverTimeObject.serverTime;
            Logger.Info("server time: {}", time);

            //get algo settings
            string exchResponse = api.get("/exchange/api/v2/info/status");
            Console.WriteLine("[[["+exchResponse+"]]]");
            DataTable exchArray = Newtonsoft.Json.JsonConvert.DeserializeObject<DataTable>(exchResponse);

            DataRow mySettings = null;
            foreach (DataRow symbol in exchArray.Rows)
            {
                if (symbol["baseAsset"].Equals(CURRENCY_BUY))
                {
                    mySettings = symbol;
                }
            }
            Logger.Info("exchange settings: {}", mySettings["currency"]);

            //get balance
            string accountsResponse = api.get("/main/api/v2/accounting/accounts", true, time);
            DataTable accountsArray = Newtonsoft.Json.JsonConvert.DeserializeObject<DataTable>(accountsResponse);

            DataRow myBalace = null;
            foreach (DataRow account in accountsArray.Rows)
            {
                if (account["currency"].Equals(CURRENCY_SELL))
                {
                    myBalace = account;
                }
            }
            Logger.Info("balance: {} {}", myBalace["balance"], CURRENCY_SELL);

            //get order book
            string orderBookResponse = api.get("/exchange/api/v2/orderbook?market=" + CURRENCY_BUY + CURRENCY_SELL + "&limit=100", true, time);
            OrderBooks orderBooks = Newtonsoft.Json.JsonConvert.DeserializeObject<OrderBooks>(orderBookResponse);
            Logger.Info("cheapest offer price: {} supply: {}", orderBooks.sell[0][0], orderBooks.sell[0][1]);

            double qty = 0.1 * 2;
            string sQty = qty.ToString("0.00000000", System.Globalization.CultureInfo.InvariantCulture);
            string sPrice = orderBooks.sell[0][0].ToString("0.00000000", System.Globalization.CultureInfo.InvariantCulture);

            //buy with limit order
            string url = "/exchange/api/v2/order?market=" + CURRENCY_BUY + CURRENCY_SELL + "&side=buy&type=limit&quantity=" + sQty + "&price=" + sPrice;
            Logger.Info("order url: {}", url);
            string orderCreateResponse = api.post(url, null, time, true);
            Logger.Info("order create: {}", orderCreateResponse);
        }
    }

}
